src/commonTest/kotlin/EscapeAppendTest.kt

```diff
@@ -0,0 +1,33 @@
+import kotlinx.html.stream.escapeAppend
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class EscapeAppendTest {
```

Comment:
Please add edge-case tests for:
- `"\\&&"`  -> `"&&amp;"`
- `"\\&amp;"` -> `"&amp"`
- `"text\\"` -> throws exception

Also, it would be good to test regular non-escaped characters. Like:
- `"\\a"`  -> `"a"`

If this is expected.

And maybe more I missed?